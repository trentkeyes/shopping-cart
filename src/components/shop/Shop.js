import React, { useState } from 'react';

import { Link } from 'react-router-dom';
import Item from './Item';

const Shop = () => {
  const [items, setItems] = useState(['abbey road', 'revolver', 'rubber soul']);

  const itemsElements = items.map((item, index) => {
    return <Item key={index} item={item} />;
  });

  return (
    <div>
      <h1>Hello from Shop</h1>
      <Link to="/">Home</Link>
      <div>{itemsElements}</div>
    </div>
  );
};

export default Shop;
