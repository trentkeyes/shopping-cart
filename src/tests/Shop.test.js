import React from 'react';
import { render, screen } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Shop from '../components/shop/Shop';
// import shopData from '../shopData';

const shopData = [
  {
    album: 'Giving You the Best That I Got',
    artist: 'Anita Baker',
    art: '/assets/album-art/GivingYoutheBestThatIGot.jpg',
    price: 9.99,
  },
  {
    album: "Don't Be Cruel",
    artist: 'Bobby Brown',
    art: '/assets/album-art/Cruelalbum.jpg',
    price: 10.99,
  },
  {
    album: 'Appetite for Destruction',
    artist: "Guns N' Roses",
    art: '/assets/album-art/GunsnRosesAppetiteforDestructionalbumcover.jpg',
    price: 12.99,
  },
];

describe('Shop component', () => {
  it('renders the available items in the store', () => {
    const shop = TestRenderer.create(<Shop shopData={shopData} />);
    expect(shop).toMatchSnapshot();
  });
});
