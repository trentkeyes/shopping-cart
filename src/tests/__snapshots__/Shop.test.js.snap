// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders the available items in the store 1`] = `
<div
  className="shop__background"
>
  <div
    className="container"
  >
    <div
      className="shop__items__flex"
    >
      <div
        className="item__card"
      >
        <div
          className="item__img"
        >
          <img
            alt="album cover"
            className="item__img"
            src="/assets/album-art/GivingYoutheBestThatIGot.jpg"
          />
        </div>
        <h3>
          Giving You the Best That I Got
        </h3>
        <h3>
          Anita Baker
        </h3>
        <h3>
          $
          9.99
        </h3>
        <div
          className="item__inputs"
        >
          <input
            className="item__quantity"
            min="1"
            name="item-quantity"
            onChange={[Function]}
            type="number"
            value={1}
          />
          <button
            className="item__add"
            onClick={[Function]}
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        className="item__card"
      >
        <div
          className="item__img"
        >
          <img
            alt="album cover"
            className="item__img"
            src="/assets/album-art/Cruelalbum.jpg"
          />
        </div>
        <h3>
          Don't Be Cruel
        </h3>
        <h3>
          Bobby Brown
        </h3>
        <h3>
          $
          10.99
        </h3>
        <div
          className="item__inputs"
        >
          <input
            className="item__quantity"
            min="1"
            name="item-quantity"
            onChange={[Function]}
            type="number"
            value={1}
          />
          <button
            className="item__add"
            onClick={[Function]}
          >
            Add to Cart
          </button>
        </div>
      </div>
      <div
        className="item__card"
      >
        <div
          className="item__img"
        >
          <img
            alt="album cover"
            className="item__img"
            src="/assets/album-art/GunsnRosesAppetiteforDestructionalbumcover.jpg"
          />
        </div>
        <h3>
          Appetite for Destruction
        </h3>
        <h3>
          Guns N' Roses
        </h3>
        <h3>
          $
          12.99
        </h3>
        <div
          className="item__inputs"
        >
          <input
            className="item__quantity"
            min="1"
            name="item-quantity"
            onChange={[Function]}
            type="number"
            value={1}
          />
          <button
            className="item__add"
            onClick={[Function]}
          >
            Add to Cart
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
